python-frontend/src/main/java/org/sonar/python/cfg/ControlFlowGraphBuilder.java

```diff
@@ -136,7 +136,10 @@ private PythonCfgBlock build(Statement statement, PythonCfgBlock currentBlock) {
       case WITH_STMT:
         return buildWithStatement((WithStatement) statement, currentBlock);
       case CLASSDEF:
-        return build(((ClassDef) statement).body().statements(), currentBlock);
+        ClassDef classDef = (ClassDef) statement;
+        PythonCfgBlock block = build(classDef.body().statements(), currentBlock);
+        block.addElement(classDef.name()); // represents binding to class name
+        return block;
```

Comment:
Check if InfiniteRecursionCheck#RecursiveCallCollector#visitClassDef should be added.